ytypes/binary_type.go

```diff
@@ -58,21 +61,24 @@ func validateBinarySlice(schema *yang.Entry, value interface{}) error {
 	}
 
 	// Check that type of value is the type expected from the schema.
-	slice, ok := value.([][]byte)
-	if !ok {
-		return fmt.Errorf("non []string type %T with value: %v for schema %s", value, value, schema.Name)
+	t := reflect.TypeOf(value)
+	if t == nil || t.Kind() != reflect.Slice || t.Elem().Name() != ygot.BinaryTypeName {
+		return fmt.Errorf("non []Binary type %T with value: %v for schema %s", value, value, schema.Name)
 	}
 
 	// Each slice element must be valid and unique.
-	tbl := make(map[string]bool, len(slice))
-	for i, val := range slice {
-		if err := validateBinary(schema, val); err != nil {
+	v := reflect.ValueOf(value)
+	tbl := make(map[string]bool, v.Len())
+	for i := 0; i < v.Len(); i++ {
+		val := v.Index(i)
+		if err := validateBinary(schema, val.Interface()); err != nil {
 			return fmt.Errorf("invalid element at index %d: %v", i, err)
 		}
-		if tbl[string(val)] {
-			return fmt.Errorf("duplicate binary type: %v for schema %s", val, schema.Name)
+		binaryVal := val.Bytes()
+		if tbl[string(binaryVal)] {
```

Comment:
```golang
if tbl[string(val.Bytes())] { ...
```

Reply:
I had done assignment to `binaryVal` since there is a few more lines of logic that use the value.
Let me know if the preference would be to just use `val.Bytes()` in the corresponding locations.

Reply:
Ah, sorry, missed that you used it again. LGTM.